setup.py

```diff
@@ -112,8 +112,19 @@ def build_extensions(self):
             VERSION = line.strip().split()[-1][1:-1]
             break
 
-with open('requirements/default.txt') as fid:
-    INSTALL_REQUIRES = [l.strip() for l in fid.readlines() if l]
+
+def parse_requirements_file(filename):
+    with open(filename) as fid:
+        requires = [l.strip() for l in fid.readlines() if l]
+
+    return requires
+
+
+INSTALL_REQUIRES = parse_requirements_file('requirements/default.txt')
+extras_require = {
+    dep : parse_requirements_file('requirements/' + dep + '.txt')
+    for dep in ['docs', 'optional', 'test']
```

Comment:
Perhaps, a comment to say that extra is omitted on purpose.

Reply:
I hope that one helps! thanks for the input.